homeassistant/components/denonavr/media_player.py

```diff
@@ -333,17 +333,17 @@ def state(self) -> MediaPlayerState | None:
         return DENON_STATE_MAPPING.get(self._receiver.state)
 
     @property
-    def source_list(self):
+    def source_list(self) -> list[str]:
         """Return a list of available input sources."""
-        return self._receiver.input_func_list
+        return self._receiver.input_func_list  # type: ignore[no-any-return]
```

Comment:
So, one thing we did in the past, is not accepting these ignores. It will riddle our codebase with ignores?

What would be the benefit of doing this everywhere?

Reply:
I understand your concern. From my POV this is the best option besides just disabling `no-any-return`.

There are other workarounds. (1) It could be assigned to a typed variable
```py
source_list: list[str] = self._receiver.input_func_list
return source_list
```
(2) `cast` would also work
```py
return cast(list[str], self._receiver.input_func_list)
```

However, both of these do come with small / tiny runtime costs. More importantly IMO is to consider what happens if / when the upstream library adds `py.typed`. Mypy could notice the unnecessary `cast` if the upstream type is the same - if it's different, e.g. the upstream one is optional, this would fail silently. The type assignment would create an error if it's different but would stay in place otherwise.

It's better to see the mypy errors for an unnecessary `type: ignore` and just remove it then IMO, even if it might not look good at the moment. You'll also quickly notice all the improvements once typing is added upstream.

Reply:
Well I agree from those options, this is the least worse.

I'm saying something else: Why enable strict typing at all in this case? Why do that effort is the whole upstream isn't typed at all?

Reply:
It helps with overall type checking, internal consistency, and intellisense support. Pylance does work even if the types aren't "officially" exported. I believe mypy does have an option for that as well but that would likely cause to many issues at the moment.

Technically speaking, this PR also fixes a small bug: `parameter.startswith` instead of `parameter.startwith`. I noticed that one because I added the type hints.

Reply:
It should also help with code reviews. If you know the input types is easier to check if a functions is doing what it should.

Reply:
How does it help, if all is ignored? It right now possibly provides an fake sense of truth.

Strict typing is part of the quality scale, most to get upstream types as well. This PR fully defeats the goal.

I think this needs broader discussion, as I think the only thing it adds: riddling our code with ignores.

Reply:
I tend to disagree but tbh I don't have an interest in a lengthly discussion about that atm. I'm happy to not continue with these PRs to enable strict typing. With this one, there are 4 of my strict-typing PRs left. Do with them what you like.
- #106822
- #106905
- #106906

Reply:
The aosmith one exposes a similar issue as this PR IMHO.

I will pick up the discussion and get back to you on Thursday

Reply:
Why does the ignore needs to be added in the first place?
There is a matching type annotation for this method ([ref](https://github.com/ol-iver/denonavr/blob/ae53172caf90c8d695087356873603dd1449b994/denonavr/denonavr.py#L337)).

Reply:
> Why does the ignore needs to be added in the first place?

The library doesn't include a 'py.typed' file. So Mypy won't use any of them as it will be considered untyped.